server/command.go

```diff
@@ -209,14 +220,26 @@ func (p *Plugin) ExecuteCommand(_ *plugin.Context, args *model.CommandArgs) (*mo
 		return &model.CommandResponse{}, nil
 	}
 
-	isSysadmin, err := p.hasSysadminRole(args.UserId)
-	if err != nil {
-		p.postCommandResponse(args, "authorization failed: %s", err)
-		return &model.CommandResponse{}, nil
-	}
-	if !isSysadmin {
-		p.postCommandResponse(args, "/welcomebot commands can only be executed by the user with system admin role")
-		return &model.CommandResponse{}, nil
+	if action == commandTriggerSetChannelWelcome || action == commandTriggerGetChannelWelcome || action == commandTriggerDeleteChannelWelcome {
+		isChannelAdmin, err := p.hasChannelAdminRole(args.ChannelId, args.UserId)
+		if err != nil {
+			p.postCommandResponse(args, "authorization failed: %s", err)
+			return &model.CommandResponse{}, nil
+		}
+		if !isChannelAdmin {
+			p.postCommandResponse(args, "/%s commands can only be executed by the user with channel admin role", action)
+			return &model.CommandResponse{}, nil
+		}
+	} else {
+		isSysadmin, err := p.hasSysadminRole(args.UserId)
+		if err != nil {
+			p.postCommandResponse(args, "authorization failed: %s", err)
+			return &model.CommandResponse{}, nil
+		}
+		if !isSysadmin {
+			p.postCommandResponse(args, "/welcomebot commands can only be executed by the user with system admin role")
+			return &model.CommandResponse{}, nil
+		}
 	}
```

Comment:
```suggestion
	if action == commandTriggerSetChannelWelcome || action == commandTriggerGetChannelWelcome || action == commandTriggerDeleteChannelWelcome {
		isChannelAdmin, err := p.hasChannelAdminRole(args.ChannelId, args.UserId)
		if err != nil {
			p.postCommandResponse(args, "authorization failed: %s", err)
			return &model.CommandResponse{}, nil
		}
		if !isChannelAdmin {
			p.postCommandResponse(args, "/%s commands can only be executed by the user with channel admin role", action)
			return &model.CommandResponse{}, nil
		}
	} 
	
	isSysadmin, err := p.hasSysadminRole(args.UserId)
	if err != nil {
		p.postCommandResponse(args, "authorization failed: %s", err)
		return &model.CommandResponse{}, nil
	}
	if !isSysadmin {
		p.postCommandResponse(args, "/welcomebot commands can only be executed by the user with system admin role")
		return &model.CommandResponse{}, nil
	}
```

Reply:
@maisnamrajusingh I have a simple request to remove the `else` block. Since all of the `if` statements include a `return` the else is not necessary.